event-bus/distributed/src/main/java/org/apache/james/events/KeyRegistrationHandler.java

```diff
@@ -128,7 +128,7 @@ private void declareQueue(Sender sender) {
             QueueSpecification.queue(registrationQueue.asString())
                 .durable(configuration.isEventBusNotificationDurabilityEnabled())
                 .exclusive(!EXCLUSIVE)
-                .autoDelete(AUTO_DELETE)
+                .autoDelete(!AUTO_DELETE)
```

Comment:
the key queue seems volatile and can accept message lost to me -> Maybe we do not need to apply quorum queue on key queues?

After James restarts, a key queue would not be consumed anymore. So I think we should keep autoDelete if possible to avoid flushing RabbitMQ with unused key queues.

Reply:
But thinking more about that, not applying quorum queue to the key queues could again break James if a RabbitMQ node is down...

Reply:
Likely we can write a task to clean up inactive RabbitMQ queues (no consumers)

Reply:
Hmm normally key queues should be cleaned up by James upon James shutdown: https://github.com/apache/james-project/blob/master/event-bus/distributed/src/main/java/org/apache/james/events/KeyRegistrationHandler.java#L140